tests/conftest.py

```
@@ -24,6 +38,53 @@
+    from loguru import logger
+
+    real_remove = logger.remove  # bind before a test can monkeypatch it to a no-op
+    before = set(logger._core.handlers)
+    try:
+        yield
+    finally:
+        for handler_id in set(logger._core.handlers) - before:
+            try:
+                real_remove(handler_id)
+            except ValueError:
+                pass  # already removed (e.g. by the command's own call_on_close)
```

Ruff (0.15.14)

[warning] Use `contextlib.suppress(ValueError)` instead of `try`-`except`-`pass`

(SIM105)